### ISSUE TITLE
Panic (sometimes) when using `:cd` without the companion shell function

### ISSUE THREAD

**Issue by jonas-schievink** [Canop/broot#4]
When exiting broot using `:c`, it outputs the selected path to stdout, but it might additionally print a (partial) panic message because an internal thread appears to panic:

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', libcore/result.rs:1009:5
```

Unfortunately I can't get a backtrace since broot exits *during* the panic.

Fixing this is probably only a matter of ignoring this kind of error or exiting the threads in a different fashion that can't cause this.

**Comment by jonas-schievink** [comment]
No, I didn't yet set that up. It indeed stops happening as soon as `--out` is passed to broot, which seems a bit odd.

**Comment by jonas-schievink** [comment]
Oh and the error also doesn't even break anything, I just though I'd still report it just to be sure.

**Comment by Canop** [comment]
You did well.

For more explanation: https://gitlab.redox-os.org/redox-os/termion/issues/163

**Comment by Canop** [comment]
I don't close this issue because a panic isn't the right behaviour, even when not using the shell function.

This being said, I don't know how to reproduce it :(

**Comment by jonas-schievink** [comment]
Yes, this might be platform-dependent due to thread scheduling details. I think it could be fixed by not unwrapping any channel receive operations, though.

**Comment by ghuls** [comment]
I think the `:c` function should check to see if the selected entry is a file or a directory.
If it is a file, it should `cd` to the parent directory. If it is a directory, it should `cd` to that  instead.

Now I get the same panic as described above when I look for `some_file` and then do `:c` (where the selected entry is a file. When the selected entry is a dir, there is not panic, and everything works.
```
$ br

some_file:c

/tmp/some_file
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by Canop** [comment]
I think the panic part has been fixed with commit 8aa96fb
